src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6278,17 +6290,30 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
                             lastTokenOfType = this.EatToken();
                             return ScanTypeFlags.NotType;
                         }
+                        else if (tupleElementType == ScanTypeFlags.MustBeType)
+                        {
+                            mustBeType = true;
```

Comment:
Turns out that's a bad idea. This method needs to scan through tokens and the caller needs to check the tokens that come after the tuple. Reverting.